### PR TITLE
point features will save as rasters

### PR DESCRIPTION
**Pull request by asaxton**

### CODE REVIEW

**Comment by asaxton** [comment]
Tested with 

```
apptainer run --nv -B /projects/bbym/saxton/MockValData:/data -B ./feedback:/feedback -B ./logs:/logs -B ./output:/output ./criticalmaas-pipeline_pr-6.sif -v --data /data/validation --output /output --legends /data/validation --log /logs/log.log --model golden_muscat --validation /data/validation_labels --output_types raster_masks
```

and 

```
apptainer run --nv -B /projects/bbym/saxton/MockValData:/data -B ./feedback:/feedback -B ./logs:/logs -B ./output:/output ./criticalmaas-pipeline_pr-6.sif -v --data /data/validation --output /output --legends /data/validation --log /logs/log.log --model flat_iceberg --validation /data/validation_labels --output_types raster_masks
```

Exited cleanly, no exceptions raised. f1 scores were `KY_WestFranklin: nan, OR_Carlton: 0.40` and `KY_WestFranklin: 0.27 OR_Carlton: 0.45` respectively. Note: `KY_WestFranklin` had no point features. f1 of nan is correct.

File: src/pipeline_steps.py [hunk 1]
```diff
@@ -227,10 +227,22 @@ def save_output(data_id, map_data: CMAAS_Map, output_dir, feedback_dir, output_t
...
+            if feature.type == MapUnitType.POINT:
+                if map_data.point_segmentation_mask is None:
+                    continue
+                feature_mask = np.zeros_like(map_data.point_segmentation_mask, dtype=np.uint8)
+                feature_mask[map_data.poly_segmentation_mask == legend_index] = 1
+                filepath = os.path.join(output_dir, sanitize_filename(f'{map_data.name}_{feature.label}.tif'))
+                io.saveGeoTiff(filepath, feature_mask, map_data.georef.crs, map_data.georef.transform)
+            if feature.type == MapUnitType.POLYGON:
+                if map_data.poly_segmentation_mask is None:
+                    #pipeline_manager.log(logging.WARNING, f'Can\'t validate feature {feature.label}. No predicted segmentation mask present.')
```
Comment (robkooper) [inline src/pipeline_steps.py:240]: You can use both types of quotes in this case.
```suggestion
                    #pipeline_manager.log(logging.WARNING, f"Can't validate feature {feature.label}. No predicted segmentation mask present.")
```

File: src/pipeline_steps.py [hunk 2]
```diff
@@ -227,10 +227,22 @@ def save_output(data_id, map_data: CMAAS_Map, output_dir, feedback_dir, output_t
     if 'raster_masks' in output_types:
         legend_index = 1
         for feature in map_data.legend.features:
-            feature_mask = np.zeros_like(map_data.poly_segmentation_mask, dtype=np.uint8)
-            feature_mask[map_data.poly_segmentation_mask == legend_index] = 1
-            filepath = os.path.join(output_dir, sanitize_filename(f'{map_data.name}_{feature.label}.tif'))
-            io.saveGeoTiff(filepath, feature_mask, map_data.georef.crs, map_data.georef.transform)
+            pipeline_manager.log(logging.DEBUG, f'{map_data.name}, {feature.label} {feature.type}')
```
Comment (robkooper) [inline src/pipeline_steps.py:230]: do we still need this line?

File: src/pipeline_steps.py [hunk 3]
```diff
@@ -227,10 +227,22 @@ def save_output(data_id, map_data: CMAAS_Map, output_dir, feedback_dir, output_t
...
+            if feature.type == MapUnitType.POINT:
```
Comment (robkooper) [inline src/pipeline_steps.py:231]: looks like most of the code is the same would it make sense to just use:
```suggestion
            if feature.type == MapUnitType.POINT or feature.type == MapUnitType.POLYGON:
```

**Review by robkooper — APPROVED** [review]
LGTM some small comments.

File: src/pipeline_steps.py (hunk 3 above)
Comment (asaxton) [reply src/pipeline_steps.py:233]: There was a small difference between the two blocks. `map_data.poly_segmentation_mask` vs. `map_data.point_segmentation_mask`. But there was redundant code in them too. Take a look now, I've made some adjustments.

**Review by robkooper — APPROVED** [review]

**Pull request merged**